### ISSUE TITLE
gevent.pool.Pool().imap_unordered hangs with an empty iterator

### ISSUE THREAD

**Issue by denik** [gevent/gevent#138]
<b>What steps will reproduce the problem?</b>

&gt;&gt;&gt; for x in Pool().imap_unordered(lambda x: x, iter([])):
...     print x
... 

<b>What is the expected output? What do you see instead?</b>
The excpected output is nothing, but instead the interpreter hangs.

<b>What version of the gevent are you using?</b>
0.13.7

<b>On what operating system?</b>
CentOS 6

<b>On what Python?</b>
2.7.3

<b>Please provide any additional information below.</b>
Occurs both running in a shell and from a script.<p><a href="http://code.google.com/p/gevent/issues/detail?id=138">Reported by eproxus.</a></p><hr/><h2>earlier comments</h2>

<p><strong>petrasync said, at 2012-06-19T15:44:02.000Z:</strong></p>

<p>The IMapUnordered class which is behind the scenes for Pool.imap_unordered() uses an underlying gevent Queue to implement the iterator protocol. Only when the iterable is non-empty and fully processed will _on_result() put a StopIteration into the queue to signal the end of results. When the iterable is empty, the iteration hangs waiting on the queue.

Fixed in a newer version:

changeset:   1569:9f605d3e437f
user:        Denis Bilenko <denis.bilenko@gmail.com>
date:        Mon Nov 01 17:47:49 2010 +0600
summary:     pool: proper implementation of imap_unordered
</p><p><strong>Denis.Bilenko said, at 2012-06-19T17:01:59.000Z:</strong></p>

<p>Thanks for the report! Fixed in https://bitbucket.org/denis/gevent/changeset/d0cf5d8a4ce0</p><p><strong>eproxus said, at 2012-06-20T07:08:43.000Z:</strong></p>

<p>Any estimate on when this fix will be in PyPi?</p><p><strong>Denis.Bilenko said, at 2012-07-26T11:27:05.000Z:</strong></p>

<p>Issue 142 has been merged into this issue.</p>
